### PR TITLE
[Merged by Bors] - feat(library/init/util): `try_for_time` interruption function

### PR DESCRIPTION
**Pull request by jesse-michael-han**
Exposes `try_for_time`, which is similar to `try_for` except that it will interrupt after `max` milliseconds have passed. This is useful because there is quite a bit of variance in the number of heartbeats used by hanging/looping tactics.

Co-author: @gebner

### CODE REVIEW

File: library/init/meta/tactic.lean [hunk 1]
```diff
@@ -1648,6 +1648,16 @@ match _root_.try_for max (tac s) with
 | none   := mk_exception "try_for tactic failed, timeout" none s
 end
 
+/-- Execute tac for 'max' milliseconds. Useful due to variance
```
Comment (bryangingechen) [inline library/init/meta/tactic.lean:1651]:
```suggestion
/-- Execute `tac` for `max` milliseconds. Useful due to variance
```

File: library/init/util.lean [hunk 2]
```diff
@@ -34,6 +34,14 @@ f ()
   This is a deterministic way of interrupting long running tasks. -/
 meta def try_for {α : Type u} (max : nat) (f : thunk α) : option α :=
 some (f ())
+
+/--
+  This function has a native implementation where
+  the thunk is interrupted if it takes more than 'max' milliseconds to compute it.
```
Comment (bryangingechen) [inline library/init/util.lean:40]:
```suggestion
  the thunk is interrupted if it takes more than `max` milliseconds to compute it.
```

File: src/library/vm/vm.cpp [hunk 3]
```diff
@@ -2389,9 +2389,25 @@ optional<vm_obj> vm_state::try_invoke_catch(vm_obj const & fn, unsigned nargs, v
             while (m_call_stack.size() > call_stack_sz) m_call_stack.pop_back();
         }
         return optional<vm_obj>();
+    } catch (lean::interrupted) {
```
Comment (gebner) [inline src/library/vm/vm.cpp:2392]:
```suggestion
    } catch (lean::interrupted &) {
```

File: src/library/vm/vm_aux.cpp [hunk 4]
```diff
@@ -56,13 +56,40 @@ vm_obj vm_try_for(vm_obj const &, vm_obj const & n, vm_obj const & thunk) {
     }
 }
 
+vm_obj vm_try_for_time (vm_obj const &, vm_obj const & n, vm_obj const & thunk) {
```
Comment (gebner) [inline src/library/vm/vm_aux.cpp:59]: You also need to provide a (dummy / always throwing an exception / always executing the thunk) implementation for the javascript version.
```suggestion
vm_obj vm_try_for_time (vm_obj const &, vm_obj const & n, vm_obj const & thunk) {
#if defined(LEAN_MULTI_THREAD)
```

**Review by bryangingechen — APPROVED** [review]
LGTM but I defer to @gebner for the C++.

**Pull request closed without merging**